lib/hal/video.c

```diff
@@ -230,6 +234,27 @@ BOOLEAN XVideoListModes(VIDEO_MODE *vm, int bpp, int refresh, void **p)
 		if(pVidMode->refresh != refresh)
 			continue;
 
+		if(dwAdapter == AV_PACK_HDTV)
+		{
+			bool is_hd = pVidMode->dwMode & 0x80000000;
+
+			/* If 480p support is enabled, reject 480i modes
+			   (forces a 480p mode to be picked) */
+			if(allow_480p && (pVidMode->height == 480 && !is_hd))
```

Comment:
What if an application exists that should force 480i mode? For example, testing how a display deals with interlacing in something similar to 240p test suite.

This would make that impossible without manually setting video mode,  as nxdk would always force the application into progressive-scan.

Although this would have little use outside of that kind of 'display test' application, so it might not be worth worring about

Reply:
The performance shouldn't be impacted when switching between 480i and 480p.
So generally the user should expect to get the best quality that their display supports (which is set up in the dashboard).

> Although this would have little use outside of that kind of 'display test' application, so it might not be worth worring about

Agreed; I assume that we'll see another API to force (or search for) possible modes explicitly, but that's a niche use-case that we should ignore for now.